R/pin_versions.R

```diff
@@ -160,11 +160,18 @@ version_from_path <- function(x) {
   out
 }
 
-version_setup <- function(board, name, new_version, versioned = NULL) {
+version_setup <- function(board, name, new_version, versioned = NULL, call = caller_env()) {
   if (pin_exists(board, name)) {
     versions <- pin_versions(board, name)
     old_version <- versions$version[[1]]
     n_versions <- nrow(versions)
+    if (old_version == new_version) {
+      cli::cli_abort(c(
```

Comment:
I definitely agree that how we make a version/hash could use some rework. I believe, though, if we do make some improvement in how we generate a data hash, we would still end up with this same problem. If you try to store a pin but the data hash hasn't changed and the timestamp hasn't changed, then we try to write to the same file in the local cache and end up with those "[EACCES] Failed to copy" errors. I don't believe making improvements to the data hashing strategy will solve this.

Reply:
If it's the same data, and less than a second has elapsed, why not just return a path to the existing pin?

Reply:
Let me know if I am not understanding what you are getting at, but that's basically the current situation that is leading to confusing errors; `version_setup()` returns a version/path to an existing pin but since the file is read only, it can't be overwritten and we get the "[EACCES] Failed to copy" errors with the long paths in them which folks find hard to interpret. Those errors are cropping up when `version_setup()` is called from the `pin_store` method (this really just applies to `board_folder()` since anything with a remote server can't respond that fast).